package/src/components/AutoCompleteInput/AutoCompleteSuggestionCommandIcon.tsx

```diff
@@ -31,54 +17,55 @@ export const AutoCompleteSuggestionCommandIcon = ({
       },
     },
   } = useTheme();
-  switch (name) {
-    case 'ban':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <UserDelete height={16} pathFill={white} width={16} />
-        </View>
-      );
-    case 'flag':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <Flag pathFill={white} />
-        </View>
-      );
-    case 'giphy':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <GiphyIcon />
-        </View>
-      );
-    case 'imgur':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <Imgur />
-        </View>
-      );
-    case 'mute':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <Mute height={16} pathFill={white} width={16} />
-        </View>
-      );
-    case 'unban':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <UserAdd height={16} pathFill={white} width={16} />
-        </View>
-      );
-    case 'unmute':
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <Sound pathFill={white} />
-        </View>
-      );
-    default:
-      return (
-        <View style={[styles.iconContainer, { backgroundColor: accent_blue }, iconContainer]}>
-          <Lightning fill={white} size={16} />
-        </View>
-      );
-  }
+
+  const renderIcon = useCallback(
+    (name: CommandVariants) => {
+      switch (name) {
+        case 'ban':
+          return <UserDelete height={16} pathFill={white} width={16} />;
+        case 'flag':
+          return <Flag pathFill={white} />;
+        case 'giphy':
+          return <GiphyIcon />;
+        case 'imgur':
+          return <Imgur />;
+        case 'mute':
+          return <Mute height={16} pathFill={white} width={16} />;
+        case 'unban':
+          return <UserAdd height={16} pathFill={white} width={16} />;
+        case 'unmute':
+          return <Sound pathFill={white} />;
+        default:
+          return <Lightning fill={white} size={16} />;
+      }
+    },
+    [white],
+  );
+
+  const icon = renderIcon(name);
```

Comment:
Here you're wrapping `renderIcon` in a `useCallback`, only to still rerender the icon fully on every render anyway. Instead of doing it this way and bypassing useful DOM reconciliation, you can define `renderIcon` as `SuggestionsCommandIcon` and then just render it like so:

```
<SuggestionsCommandIcon />
```
without any props.

Reply:
Okay